index.js

```diff
@@ -59,15 +112,30 @@ class QuickLRU {
 	}
 
 	has(key) {
+		if (this.maxAge > 0) {
+			return Boolean(this._getOrDeleteIfExpired(
+				key,
+				this.cache.get(key) || this.oldCache.get(key)
+			));
+		}
+
 		return this.cache.has(key) || this.oldCache.has(key);
 	}
 
 	peek(key) {
 		if (this.cache.has(key)) {
+			if (this.maxAge > 0) {
+				return this._getOrDeleteIfExpired(key, this.cache.get(key));
+			}
+
 			return this.cache.get(key);
 		}
 
 		if (this.oldCache.has(key)) {
+			if (this.maxAge > 0) {
+				return this._getOrDeleteIfExpired(key, this.oldCache.get(key));
+			}
+
```

Comment:
It would be great if you could go through the code and try to simplify things and DRY things up as this PR introduces a lot of boilerplate and complexity. For example, both of the instructions above are the same except for the cache, so they both could use an internal `_peek` method that accepts a cache.

Reply:
true, okey let me refactor it a little bit ;)